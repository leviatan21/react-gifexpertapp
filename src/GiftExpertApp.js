import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GiftGrid } from './components/GiftGrid';
//	rafc
export const GiftExpertApp = ( { defaultState=[] } ) => {
	//	
	const initialState = ( defaultState.length ) ? defaultState : [ 'One Punch' ];
	const [ categories, setCategories ] = useState( initialState );
	//
	return(
		<>
			<h1>GiftExpertApp</h1>
			<AddCategory setCategories={ setCategories } />
			<hr/>
			{ categories.map( category => <GiftGrid key={ category } category={ category } /> ) }
		</>
	)
};
