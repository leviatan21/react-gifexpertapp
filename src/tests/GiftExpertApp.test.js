
import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GiftExpertApp } from '../GiftExpertApp';
//
describe( 'Puebas en <GiftExpertApp />', ( ) => {
	//
	test( 'Componente GiftExpertApp', ( ) => {
		//
		const defaultState 		= [ 'Batman', 'Spiderman' ];
		const wrapper			= shallow( <GiftExpertApp defaultState={ defaultState } /> );
		const findHeader1		= wrapper.find( 'h1' );
		const findAddCategory	= wrapper.find( 'AddCategory' );
		const findGiftGrid		= wrapper.find( 'GiftGrid' );
		//
		expect( wrapper ).toMatchSnapshot( );
		expect( findHeader1.length ).toBe( 1 ); 
		expect( findAddCategory.length ).toBe( 1 ); 
		expect( findGiftGrid.length ).toBe( defaultState.length ); 
	} );

} );
