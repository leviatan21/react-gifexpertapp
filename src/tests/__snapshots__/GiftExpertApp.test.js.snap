// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Puebas en <GiftExpertApp /> Componente GiftExpertApp 1`] = `
<Fragment>
  <h1>
    GiftExpertApp
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <GiftGrid
    category="Batman"
    key="Batman"
  />
  <GiftGrid
    category="Spiderman"
    key="Spiderman"
  />
</Fragment>
`;
